### ISSUE TITLE
docker run analyze fails with CC::CLI::Analyze::EngineFailure on 0.70

### ISSUE THREAD

**Issue by gitlab-winnie** [codeclimate/codeclimate#782]
- Is this an issue with codeclimate.com?

no

- Is this an issue with a specific engine?

I don't think so.

- Is this an issue with your Docker installation in general?

no

- Are you invoking `docker run codeclimate/codeclimate` without other options?

no, with options:

https://gitlab.com/gitlab-org/gitlab-ce/blob/efe9d551988ae0566fe7a8f6759d3fa8973c3990/.gitlab-ci.yml#L581

- Are you having trouble using this project as a gem or not via Docker?

no

## Guidelines

If you'd still like to report this issue, please follow these guidelines:

- Make sure your problem exists in the latest version of the CLI and any engines
  you're using, the following commands will update everything:

The problem only exists with the latest version (0.70) and not with 0.69.

- Include the exact command you're running and its complete output

  Please enable debug for these invocations:

  ```
  CODECLIMATE_DEBUG=1 codeclimate analyze
  ```

https://gitlab.com/gitlab-org/gitlab-ce/-/jobs/40775443

- Indicate what you expected to happen and what happened instead

The command fails with

```
error: (CC::CLI::Analyze::EngineFailure) engine structure failed with status 99 and stderr 
CC::Analyzer::Formatters::JSONFormatter#write returned false, indicating an error
```

and instead we expect the command to succeed.

- Include your operating system, how you run Docker (e.g. Docker Machine) and
  the version of Docker you're using.

It is failing on different machines, so I think this is not relevant—please let me know if you need the information anyway.

- If your project is open source, please provide a URL to the source

https://gitlab.com/gitlab-org/gitlab-ce/

**Comment by gitlab-winnie** [comment]
We think (https://gitlab.com/gitlab-org/gitlab-ce/issues/40255#note_47571857) that this is caused by a config change from 0.69 to 0.70. However running `codeclimate init --upgrade` (as described in `Updating existing Code Climate configuration` on https://hub.docker.com/r/codeclimate/codeclimate/) resulted in `unknown command init`. I also could not find upgrade instructions or a changelog entry for the configuration change.

**Comment by gitlab-winnie** [comment]
Using `CODECLIMATE_DEBUG=1` the command succeeded: https://gitlab.com/gitlab-org/gitlab-ce/-/jobs/40775443 but contained multiple `CC::Engine::Analyzers::ParserError`s. So the root cause may be in our code or in the [babylon parser](https://github.com/babel/babel/tree/master/packages/babylon)

**Comment by patvdleer** [comment]
Can confirm
```
$ codeclimate version
0.70.0

$ codeclimate analyze  -f json > codeclimate.json
error: (CC::CLI::Analyze::EngineFailure) engine structure failed with status 99 and stderr 
CC::Analyzer::Formatters::JSONFormatter#write returned false, indicating an error
```

**Comment by aequitas** [comment]
Running into the same issue: https://gitlab.com/failmap/admin/-/jobs/40729986

**Comment by frenck** [comment]
Having the same issue in our GitLab CI.

Enforced the use of CodeClimate 0.69.0 for now, until this is fixed.

**Comment by brammittendorff** [comment]
Same issues here in our GitLab CI. Confirmed.

**Comment by mjrider** [comment]
Same here, broken Gitlab-CI builds

**Comment by toddmazierski** [comment]
Thanks for the report! We're looking into the issue now.

**Comment by toddmazierski** [comment]
My apologies for the trouble, all. A fix is included in release [v0.70.1](https://github.com/codeclimate/codeclimate/releases/tag/v0.70.1) if you wouldn't mind upgrading. Thanks!

**Comment by toddmazierski** [comment]
One note about the fix: there is a new `type` of document included in the JSON output named `measurement`, described here: codeclimate/spec#59.

**Comment by gitlab-winnie** [comment]
@toddmazierski Thank you very much for the fast reaction! 🚀 

I can confirm that this is fixed with 0.70.1: https://gitlab.com/gitlab-org/gitlab-ce/-/jobs/40869216

**Comment by frenck** [comment]
👍  Thanks for the quick fix!
